### PR TITLE
feat: Add imagesDimensions field.

### PR DESCRIPTION
**Pull request by dohooo**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 3fc91b6</samp>

This pull request adds functionality to detect and store the dimensions of images that are part of the metadata content for pages and feed items. It uses the `image-size` package as a dependency and modifies the database schema, the Prisma schema, the API routes, and the types to handle the image dimensions. It also adds a helper function `decoratePageForImageDimensions` to fetch and add the dimensions from the image files.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 3fc91b6</samp>

> _We are the masters of the image size_
> _We know the width and height of every file_
> _We store the dimensions in the JSONB_
> _We decorate the pages with the metadata_

### WHY

Now, The `/api/feed` and `/api/pages` api could include the image dimensions field. The client could use this field to better display the cover image in some situations, such as  in the masonry list.

<details>
  <summary>Caching image dimensions</summary>
image
</details>

#### `/api/feed` Fetching result:
```json
{
    "list": [
        {
            "metadata": {
                "uri": "ipfs://QmfH9THegRpjAdyGheK36qPnJ6rVPnQaft1vQwB8LGj1R3",
                "content": {
                    "tags": [
                        "post",
                        "starknet"
                    ],
                    "type": "note",
                    "title": "StarkNet 代币释放在即，其主网数据表现如何？",
                    "sources": [
                        "xlog"
                    ],
                    "summary": "“区块链三难题”，或 “可扩展性三难题”，强调了区块链平台想要去平衡安全性、去中心化和可扩展性将面临的挑战。通常情况下，区块链架构只能有效地优先考虑其中两个难题。例如，以太坊优先考虑了安全性和去中心化，导致了可扩展性方面面临着许多挑战。但以太坊社区已经认识到可扩展性的重要性…",
                    "attributes": [
                        {
                            "value": "StarkNet-dai-bi-shi-fang-zai-ji--qi-zhu-wang-shu-ju-biao-xian-ru-he-",
                            "trait_type": "xlog_slug"
                        }
                    ],
                    "attachments": [
                        {
                            "name": "cover",
                            "address": "ipfs://QmV263n5pNv8XTfASLeFLiAzmnaLCjPoBd2Dc7QrfAequo",
                            "mime_type": "image/png"
                        }
                    ],
                    "date_published": "2023-09-18T16:00:00.000Z",
                    "audio": "",
                    "frontMatter": {},
                    "cover": "ipfs://QmV263n5pNv8XTfASLeFLiAzmnaLCjPoBd2Dc7QrfAequo",
                    "images": [
                        "ipfs://QmV263n5pNv8XTfASLeFLiAzmnaLCjPoBd2Dc7QrfAequo"
                    ],
                    "slug": "StarkNet-dai-bi-shi-fang-zai-ji--qi-zhu-wang-shu-ju-biao-xian-ru-he-",
                    "disableAISummary": false,
                    "readingTime": 5,
                    "imageDimensions": { // New field
                        "ipfs://QmV263n5pNv8XTfASLeFLiAzmnaLCjPoBd2Dc7QrfAequo": {
                            "width": 755,
                            "height": 429,
                            "uri": "https://ipfs.4everland.xyz/ipfs/QmV263n5pNv8XTfASLeFLiAzmnaLCjPoBd2Dc7QrfAequo"
                        }
                    }
                }
            },
            "characterId": 59765,
            "noteId": 10,
            "character": {
                "handle": "486-5551",
                "characterId": "59765",
                "metadata": {
                    "content": {
                        "bio": "hi",
                        "name": "486",
                        "type": "character",
                        "avatars": [
                            "ipfs://QmQ28CvbppVkNPq6VoU8uGM1gbKKFtb6KVtwqwnRfymUpz"
                        ]
                    }
                }
            },
            "createdAt": "2023-11-22T15:08:48.000Z"
        }
    ],
    "cursor": "59765_10",
    "count": 1
}
```

#### `/api/pages` Fetching result:
```json
{
    "list": [
        {
            "metadata": {
                "uri": "ipfs://QmYswh25pPx9amgzSD1CKXtcG1mV2vDsiRVBYEcW55Nf6b",
                "content": {
                    "tags": [
                        "short"
                    ],
                    "type": "note",
                    "title": "通往公园的路口",
                    "sources": [
                        "xlog"
                    ],
                    "attributes": [
                        {
                            "value": "6XMCfqZiVE64VzMtUsHhY",
                            "trait_type": "xlog_slug"
                        }
                    ],
                    "attachments": [
                        {
                            "name": "image",
                            "address": "ipfs://QmZwwhc2qAGPHVeCY83ETUnuoABBJMqvU4nAjUzv3Mx6VN",
                            "mime_type": "image/jpeg"
                        }
                    ],
                    "date_published": "2023-11-21T03:39:23.774Z",
                    "summary": "➡️",
                    "audio": "",
                    "frontMatter": {},
                    "cover": "",
                    "images": [
                        "ipfs://QmZwwhc2qAGPHVeCY83ETUnuoABBJMqvU4nAjUzv3Mx6VN"
                    ],
                    "slug": "6XMCfqZiVE64VzMtUsHhY",
                    "disableAISummary": false,
                    "readingTime": 0,
                    "imageDimensions": { // New field
                        "ipfs://QmZwwhc2qAGPHVeCY83ETUnuoABBJMqvU4nAjUzv3Mx6VN": {
                            "width": 960,
                            "height": 1280,
                            "uri": "https://ipfs.4everland.xyz/ipfs/QmZwwhc2qAGPHVeCY83ETUnuoABBJMqvU4nAjUzv3Mx6VN"
                        }
                    }
                }
            },
            "characterId": 56651,
            "noteId": 7,
            "createdAt": "2023-11-21T03:39:27.000Z",
            "stat": {
                "viewDetailCount": 9,
                "commentsCount": 0,
                "likesCount": 1
            },
            "_count": {
                "fromNotes": 0,
                "links": 1
            }
        }
    ],
    "count": 1,
    "cursor": "56651_7"
}
```

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 3fc91b6</samp>

*  Add `image-size` package as a dependency to get image dimensions ([link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R86), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR167-R169), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR7570-R7577), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR10399-R10404))
*  Add `image_dimensions` column to `Metadata` table in database schema ([link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-e7df923b8f507b9ad05de48d00b787b465361f94e68e15d2f9f6fed5487dfe8dL1-L-1), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-5b443964f4f3a611682db8f7e02177b0a8c632b2039e2bd5e4dd7347815c565cR21-R29))
*  Add `imageDimensions` property to `ExpandedNote` type ([link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-b8b77f5be18cf56dca425b3a5b8e9d2e754dd37fe0e3612b95cd4e9bccda08a5R92))
*  Add functions to request, cache, and decorate image dimensions in `src/queries/page.server.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-d2337de36ce4b194d949a001541129294fd9468b84ba4a17488dadff0e35ff7dL1-R4), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-d2337de36ce4b194d949a001541129294fd9468b84ba4a17488dadff0e35ff7dL10-R13), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-d2337de36ce4b194d949a001541129294fd9468b84ba4a17488dadff0e35ff7dR41-R179))
*  Call `decoratePageForImageDimensions` function in feed and pages routes to add image dimensions to metadata content ([link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-c6ea66c9ca56c93baf029fef373c6e0daef4a78d51cc74029ea90446bba845a4R3), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-c6ea66c9ca56c93baf029fef373c6e0daef4a78d51cc74029ea90446bba845a4R20-R24), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-3f620421afee77f6fbdaed934aad13674f2cf8f228eddf0a32ab83541986c2ddR4), [link](https://github.com/Crossbell-Box/xLog/pull/1059/files?diff=unified&w=0#diff-3f620421afee77f6fbdaed934aad13674f2cf8f228eddf0a32ab83541986c2ddR24-R28))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

File: src/queries/page.server.ts [hunk 1]
```diff
@@ -35,6 +38,145 @@ export const fetchGetPage = async (
   })
 }
 
+export const getImageDimensionByUri = async (uri: string) => {
+  return new Promise<{ width: number; height: number; uri: string }>(
+    (resolve, reject) => {
+      const options = url.parse(uri)
+
+      https.get(options, function (response) {
```
Comment (DIYgod) [inline src/queries/page.server.ts:46]: It is best to use fetch here. On one hand, it is a more modern API. On the other hand, Next.js has some optimizations for fetch.

**Review by DIYgod — CHANGES_REQUESTED** [review]

File: src/queries/page.server.ts (hunk 1 above)
Comment (dohooo) [reply src/queries/page.server.ts:46]: Got it! Thanks!

File: src/lib/expand-unit.ts [hunk 2]
```diff
@@ -100,6 +102,40 @@ export const expandCrossbellNote = async ({
       ...new Set(expandedNote.metadata.content.images),
     ]
 
+    if (useScore) {
```
Comment (DIYgod) [inline src/lib/expand-unit.ts:105]: Repeat with the following code below. https://github.com/Crossbell-Box/xLog/pull/1059/files#diff-0802612f2ebfab23d2656fc2189271c4fef6451452a2bd7dacc8d8bcfe11e2e8R193

File: package.json [hunk 3]
```diff
@@ -81,6 +81,9 @@
     "fast-deep-equal": "3.1.3",
     "hast-util-from-html": "2.0.1",
     "hast-util-to-html": "9.0.0",
+    "i18next": "23.7.6",
```
Comment (DIYgod) [inline package.json:84]: redundant dependencies

**Review by DIYgod — CHANGES_REQUESTED** [review]

**Pull request merged**